client/dogs/FormPage.js

```diff
@@ -65,23 +67,46 @@ export default class IndexPage extends React.Component {
   constructor(props: Props) {
     super(props);
 
-    this.state = {
-      firstName: '',
-      lastName: '',
-      phone: '',
-      email: '',
-      address: '',
-      neighborhood: '',
-      zip: '',
-      dogName: '',
-      dogAge: '',
-      color: '',
-      breed: '',
-      sex: '',
-      rabiesIssued: '',
-      rabiesExpire: '',
-      comments: '',
-    };
+    if (props.results && props.results.length) {
+      const result = props.results[0];
+      this.state = {
+        firstName: result['firstName'] || '',
+        lastName: result['lastName'] || '',
+        phone: result['phone'] || '',
+        email: '',
+        address: result['address'] || '',
+        neighborhood: result['neighborhood'] || '',
+        zip: result['zip'] || '',
+        dogName: result['dogName'] || '',
+        dogYear: result['yearsOld'] || '',
+        dogMo: result['monthsOld'] || '',
+        color: result['priColor'] || '',
+        breed: result['priBreed'] || '',
+        sex: result['sex'] || '',
+        rabiesIssued: result['vacDate'] || '',
+        rabiesExpire: result['vacExp'] || '',
+        comments: '',
+      };
+    } else {
+      this.state = {
+        firstName: '',
+        lastName: '',
+        phone: '',
+        email: '',
+        address: '',
+        neighborhood: '',
+        zip: '',
+        dogName: '',
+        dogYear: '',
+        dogMo: '',
+        color: '',
+        breed: '',
+        sex: '',
+        rabiesIssued: '',
+        rabiesExpire: '',
+        comments: '',
+      };
+    }
   }
 
   handleInputChange = (ev: SyntheticInputEvent) => {
```

Comment:
Can shorten this slightly to:

`const {value, name} = ev.target;`

to destructure `ev.target` into those two variables.